src/views/templates/details/tabs/details/components/WorkloadProfile.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import produce from 'immer';
+import useWorkloadProfile, {
+  WorkloadProfileKeys,
+} from 'src/views/templates/list/hooks/useWorkloadProfile';
+import { ensurePath } from 'src/views/templates/utils';
+
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { getTemplateVirtualMachineObject } from '@kubevirt-utils/resources/template';
+import { VM_WORKLOAD_ANNOTATION } from '@kubevirt-utils/resources/vm';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import { PencilAltIcon } from '@patternfly/react-icons';
+
+import { TemplateDetailsGridProps } from '../TemplateDetailsPage';
+
+import WorkloadProfileModal from './WorkloadProfileModal';
+
+const WorkloadProfile: React.FC<TemplateDetailsGridProps> = ({ template, editable }) => {
+  const { createModal } = useModal();
+  const { t } = useKubevirtTranslation();
+  const workload = useWorkloadProfile(template, t);
+
+  const updateWorkload = (updatedWorkload: WorkloadProfileKeys) => {
+    const updatedTemplate = produce<V1Template>(template, (templateDraft: V1Template) => {
+      const vm = getTemplateVirtualMachineObject(templateDraft);
+      ensurePath(vm, 'spec.template.metadata.annotations');
+
+      vm.spec.template.metadata.annotations[VM_WORKLOAD_ANNOTATION] = updatedWorkload;
+
+      return templateDraft;
+    });
+
+    return k8sUpdate({
+      model: TemplateModel,
+      data: updatedTemplate,
+      ns: updatedTemplate?.metadata?.namespace,
+      name: updatedTemplate?.metadata?.name,
+    });
+  };
+
+  const onEditClick = () =>
+    createModal(({ isOpen, onClose }) => (
+      <WorkloadProfileModal
+        obj={template}
+        isOpen={isOpen}
+        onClose={onClose}
+        onSubmit={updateWorkload}
+      />
+    ));
+
+  return (
+    <DescriptionListGroup>
+      <DescriptionListTerm>{t('Workload profile')}</DescriptionListTerm>
+      <DescriptionListDescription>
+        {workload}
```

Comment:
```suggestion
        { t(workloadProfiles[workload] || 'Other')}
```
you can avoid using `t` as a method parameter ,outside a component, by sending the label key instead of the translated label, see https://github.com/kubevirt-ui/kubevirt-plugin/pull/477/files#r889840969